### ISSUE TITLE
Create RDS postgres db from existing snapshot

### ISSUE THREAD

**Issue by denmat** [convox/rack#2202]
We would like to be able to restore a database from an exisiting database snapshot. We see this as part of a normal development cycle as we as a disaster recovery process.

Naively, this _*could*_ possibly be done with the following:

```
--- a/provider/aws/templates/resource/postgres.tmpl
+++ b/provider/aws/templates/resource/postgres.tmpl
@@ -21,6 +21,11 @@
         "Default": "db.t2.micro",
         "Description": "Instance class for database nodes"
       },
+      "DBSnapshotIdentifier": {
+        "Type": "String",
+        "Default": "AWS::NoValue",
+        "Description": "ARN of database snapshot to restore from"
+      },
       "Family": {
         "Type": "String",
         "Default": "postgres9.6",
@@ -109,6 +114,7 @@
           "DBInstanceClass": { "Ref": "InstanceType" },
           "DBInstanceIdentifier": { "Ref": "AWS::StackName" },
           "DBName": { "Ref": "Database" },
+          "DBSnapshotIdentifier": { "Ref": "DBSnapshotIdentifier" },
           "DBParameterGroupName": { "Ref": "ParameterGroup" },
           "DBSubnetGroupName": { "Ref": "SubnetGroup" },
           "Engine": "postgres",
```

Happy to help out getting this in.

**Comment by mwisner** [comment]
This is something we would definitely use. 

One of the things I've never been sure about is once we restore from a snapshot now how to get Convox to know about that new postgres instance...

**Comment by mwarkentin** [comment]
Generally you manage the connection with a DATABASE_URL environment variable, no? Just need to update that to point at the new DB?

**Comment by mwisner** [comment]
The app-to-app connections were fine, but generally speaking (and this could totally be user error)

1) I couldn't figure out how to get the new postgres instances to show up as a convox resource, not from the perspective of app connections, but for usage in things like `convox resource proxy`... Using Convox resource proxy is currently our defined strategy for obtaining database dumps of development or qa environments in the event of needing to debug or application for whatever reason.

This is important for us because it's kind of a PITA to go and update the actual RDS instance to support public access. Having to mark is as public and then mess with the security groups and whatnot.

Using Convox proxy is about 100x easier =). I couldn't figure out how to "import" an existing rds instance.

2) There are a couple settings around security groups and subnets that have to ensure that the ones that were previously created were used to make sure everything is setup correctly in terms of communication with the specific rack.

I'm sure it's possible to set all of that up manually, but when first working with Convox I wasn't sure what settings were really required by convox and which weren't. Probably due to my lack of deep AWS experience. 

So when we went to go from one environment to 3, prod,qa,staging for our application, I wanted to take the existing data and basically deploy it to new racks. To do this in a way I knew would work out  I had to create a new postgres instance for the new app, then go in and delete once deleted use the leftover security groups. 

so TL;DR... i'll admit it might just be user error. But Convox is about 100x easier to use and setup than AWS itself, and I think that's part of the point. But with my personal workflows this was probably the only confusing thing we had to do in AWS for the entire app setup (Aside from having to deal with cloudformation errors)

**Comment by mwarkentin** [comment]
You can use convox proxy (not convox resource proxy) to proxy to non-convox dbs. Not quite as simple (you have to provide the full host). But you don’t need to expose the db publicly. 

Configuring  the other stuff is needed I think. We manage all of that as well as the dbs with terraform on our side.

**Comment by jontonsoup** [comment]
Seems like this is live now and can be closed?
